common/flatpak-utils.c

```diff
@@ -4210,23 +4210,38 @@ static gboolean
 copy_icon (const char        *id,
            GFile             *icons_dir,
            OstreeRepo        *repo,
+           GFile             *root,
            OstreeMutableTree *size_mtree,
            const char        *size,
            GError           **error)
 {
   g_autofree char *icon_name = g_strconcat (id, ".png", NULL);
   g_autoptr(GFile) size_dir = g_file_get_child (icons_dir, size);
   g_autoptr(GFile) icon_file = g_file_get_child (size_dir, icon_name);
+  g_autoptr(GFileInfo) file_info = NULL;
+  g_autoptr(GFile) f = NULL;
   const char *checksum;
+  const char *target_icon_name = icon_name;
 
   if (!ostree_repo_file_ensure_resolved (OSTREE_REPO_FILE(icon_file), NULL))
     {
       g_debug ("No icon at size %s for %s", size, id);
       return TRUE;
     }
 
+  f = g_file_resolve_relative_path (root, g_file_get_path(icon_file) );
```

Comment:
Why do you need to to this resolve stuff? You should be able to query-info icon_file directly, no?